dart/constraint/ConstraintSolver.hpp

```diff
@@ -259,6 +277,9 @@ class ConstraintSolver
 
   /// Constraint group list
   std::vector<ConstrainedGroup> mConstrainedGroups;
+
+  /// Factory for ContactSurfaceParams for each contact
+  ContactSurfaceHandlerPtr mContactSurfaceHandler;
```

Comment:
I think this has an ABI impact

Reply:
Do you think there is an alternative approach that wouldn't break ABI?

Reply:
I don't see any obvious solutions

Reply:
I'll sketch the standard ABI-retain hack with a global `std::map` and mutex.